distributed/utils.py

```diff
@@ -1000,3 +1001,49 @@ def __init__(self, *args, **kwargs):
 
     def emit(self, record):
         self.deque.append(record)
+
+
+class ThrottledGC(object):
+    """Wrap gc.collect to protect against excessively repeated calls.
+
+    Allows to run throttled garbage collection in the workers as a
+    countermeasure to e.g.: https://github.com/dask/zict/issues/19
+
+    collect() does nothing when repeated calls are so costly and so frequent
+    that the thread would spend more than max_in_gc_frac doing GC.
+
+    warn_if_longer is a duration in seconds (10s by default) that can be used
+    to control to log a warning level message whenever an actual call to
+    gc.collect() lasts too long.
+    """
+    def __init__(self, max_in_gc_frac=0.2, warn_if_longer=10, logger=None):
```

Comment:
Nitpicking, but allowing 20% of the time inside the GC sounds like a lot.

Reply:
Would 5% be a reasonable default?

Reply:
It certainly sounds better...